src/_guides/language/effective-dart/design.md

```diff
@@ -974,79 +1060,260 @@ var names = people.map((Person person) {
 });
 {% endprettify %}
 
+In rare cases, the surrounding context is not precise enough to provide a type
+for one or more of the function's parameters. In those cases, you may need to
+annotate.
+
 
-### AVOID annotating with `dynamic` when not required.
+### PREFER annotating with `dynamic` even when not required.
```

Comment:
I'm very glad for this switch in the guideline (it matches my intuition better).